### PR TITLE
{Service Connector} Improvements for GA

### PR DESCRIPTION
**Pull request by houk-ms**
**Description**

This PR
- ~~Adds new command group `az aks connection` to support creating aks connections with service connector~~
- Upgrades service connector's api-version to `2022-05-01`
- Removes the preview flag for `az aks connection`, `az spring-cloud connection`, `az webapp connection`
- Fixes some known issues

 
**Testing Guide**

**History Notes**

[Service Connector] `az aks connection`: Add new command group to support AKS connection

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
Service Connector

File: src/azure-cli/azure/cli/command_modules/serviceconnector/_help.py [hunk 1]
```diff
@@ -47,7 +47,7 @@ def get_auth_info_params(auth_type):
         AUTH_TYPE.Secret: '--secret name=XX secret=XX',
         AUTH_TYPE.SecretAuto: '--secret',
         AUTH_TYPE.SystemIdentity: '--system-identity',
-        AUTH_TYPE.ServicePrincipalSecret: '--service-principal client-id=XX secret=XX',
+        AUTH_TYPE.ServicePrincipalSecret: '--service-principal client-id=XX object-id=XX secret=XX',
```
Comment (yungezz) [inline src/azure-cli/azure/cli/command_modules/serviceconnector/_help.py:50]: object_id is optional right?

**Review by yungezz — APPROVED** [review]

File: src/azure-cli/azure/cli/command_modules/serviceconnector/_help.py (hunk 1 above)
Comment (houk-ms) [reply src/azure-cli/azure/cli/command_modules/serviceconnector/_help.py:50]: yes, it's optional.
- add this to work around a CLI linter bug https://github.com/Azure/azure-cli-dev-tools/issues/331
- in argument help, it's still optional
image
- it only appears in aks+keyvault command example

**Comment by houk-ms** [comment]
Hide the support of `az aks connection` command group due to AKS feature delay

**Review by kairu-ms — APPROVED** [review]

**Pull request merged**